rma/models/rma_order.py

```diff
@@ -54,7 +54,12 @@ def _compute_state(self):
             rma_line_to_approve = self.env["rma.order.line"].search_count(
                 [("id", "in", rec.rma_line_ids.ids), ("state", "=", "to_approve")]
             )
-            if rma_line_done != 0:
+            rma_line_no_cancel = rec.rma_line_ids.filtered(
```

Comment:
states = set(rec.rma_line_ids.mapped("state")

if states == {"done"}:
    rec.state = "done"
elif states == {"cancel"}:
    rec.state = "cancel"
elif "to_approved" in states:
    rec.state = "to_approved"
else:
    rec.state = "approved"

Reply:
@sebastienbeau , it's done.